src/Libraries/Microsoft.Extensions.Caching.Hybrid/Internal/DefaultJsonSerializerFactory.cs

```diff
@@ -50,11 +54,42 @@ internal static bool IsFieldOnlyType(Type type)
         return IsFieldOnlyType(type, ref state) == FieldOnlyResult.FieldOnly;
     }
 
+    private static bool IsDefaultJsonOptions(JsonSerializerOptions options)
+    {
+#if NETCOREAPP3_0_OR_GREATER || NETSTANDARD2_1_OR_GREATER
+        if (!System.Runtime.CompilerServices.RuntimeFeature.IsDynamicCodeSupported)
```

Comment:
Instead of IsDynamicCodeSupported, these checks should be based on `JsonSerializer.IsReflectionEnabledByDefault`. That's a feature switch that is disabled in both trimmed and AOT'd apps.

Reply:
Great feedback, will tweak